### PR TITLE
 [Feat] make allocator customizable

### PR DESCRIPTION
**Pull request by rhdong**

### CODE REVIEW

File: tests/merlin_hashtable_test.cc.cu [hunk 1]
```diff
@@ -53,6 +55,77 @@ struct ExportIfPredFunctor {
   }
 };
 
+class CustomizedAllocator : public virtual BaseAllocator {
```
Comment (rhdong) [inline tests/merlin_hashtable_test.cc.cu:58]: Hi @Lifann, here is a demo of the customized allocator.

**Review by Lifann — APPROVED** [review]
Numerical check passed for me.

**Review by Lifann — APPROVED** [review]
LGTM

**Pull request merged**